plugins/rest/aws.go

```diff
@@ -64,7 +65,10 @@ func (cs *awsEnvironmentCredentialService) credentials() (awsCredentials, error)
 	if creds.RegionName == "" {
 		return creds, errors.New("no " + awsRegionEnvVar + " set in environment")
 	}
-	creds.SecurityToken = "" // not applicable to this credential provider
+	// SecurityToken is required if using ENV credentials from assumed IAM role
+	// Missing SecurityToken results with 403 s3 error.
+	creds.SecurityToken = os.Getenv(securityTokenEnvVar)
+
```

Comment:
Hi, I updated docs and added test case. You can check the note on this AWS page: https://docs.aws.amazon.com/general/latest/gr/sigv4-add-signature-to-request.html SecurityToken is required when we use temporary credentials (it does not exist for long-term creds). However, based on the envs we cannot detect if the user is using long-term or temporary credentials so returning an error in case of missing security token does not make sense.